packages/mdc-chips/chip/foundation.js

```diff
@@ -121,11 +124,36 @@ class MDCChipFoundation extends MDCFoundation {
 
   /**
    * Handles a transition end event on the root element.
-   * This is a proxy for handling a transition end event on the leading icon or checkmark,
-   * since the transition end event bubbles.
    * @param {!Event} evt
    */
   handleTransitionEnd_(evt) {
+    // Handle transition end event on the chip when the it is about to be removed.
+    if (this.adapter_.eventTargetHasClass(/** @type {!EventTarget} */ (evt.target), cssClasses.CHIP_EXIT)) {
+      if (evt.propertyName === 'width') {
+        this.adapter_.removeFromDOM();
```

Comment:
Should we also be calling `this.destroy()` here? AFAIK we're never destroying these chips even though they're effectively removed.

Also, just checking, chip set isn't holding any references to entry chips, is it? (I think it only holds references to selected chips for filter/choice.)

Reply:
Oh that's a really good point, the chip set does hold a reference to all its chips, regardless of the type of chip set.

I refactored it so that the chip set is in charge of removing the chip object from its reference, and also calls `remove` on the chip, which will destroy and remove itself from the DOM.

Note that this means the chip set foundation has to pass a chip object to its adapter, and the chip object needs to have an explicit `remove` method.